core/src/server/rpc_module.rs

```diff
@@ -721,39 +561,73 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	///
 	/// # Returns
 	///
-	/// An async block which returns `Result<(), SubscriptionCallbackError>` the error is simply
-	/// for a more ergonomic API and is not used (except logged for user-related caused errors).
-	/// By default jsonrpsee doesn't send any special close notification,
-	/// it can be a footgun if one wants to send out a "special notification" to indicate that an error occurred.
+	/// An async block which returns something that implements [`crate::server::IntoSubscriptionResponse`] which
+	/// decides what action to take once the subscription callback returns i.e, whether to sent out another message
+	/// on the subscription stream before closing down it.
+	///
+	/// This is implemented for `Result<T, E>` and `Option<T>`.
+	///
+	/// It's recommended to use `Result` if you want to propagate the error as special error notification
+	/// and `Option` if no error's should be propagated.
 	///
-	/// If you want to a special error notification use `SubscriptionSink::close` or
-	/// `SubscriptionSink::send` before returning from the async block.
+	/// Another option is to implement [`crate::server::IntoSubscriptionResponse`] if you want customized behaviour.
+	///
+	/// The error notification has the following format:
```

Comment:
If I'm right above, maybe also add a note on what the error looks like if the Subscription wasn't accepted yet?

Reply:
Ah I see; it is a different thing!